src/Symfony/Component/Console/Input/Input.php

```diff
@@ -171,4 +193,55 @@ public function getStream()
     {
         return $this->stream;
     }
+
+    /**
+     * Returns a stringified representation of the options passed to the command.
+     *
+     * InputArguments MUST be escaped as well as the InputOption values passed to the command.
+     *
+     * @param string[] $optionNames Name of the options returned. If empty, all options are returned and non-passed or non-existent are ignored.
+     *
+     * @return list<string>
+     */
+    public function unparse(array $optionNames = []): array
+    {
+        $rawOptions = $this->getRawOptions();
+
+        $filteredRawOptions = 0 === \count($optionNames)
+            ? $rawOptions
+            : array_intersect_key($rawOptions, array_fill_keys($optionNames, ''),
+            );
+
+        return array_map(
+            fn (string $optionName) => $this->unparseOption(
+                $this->definition->getOption($optionName),
+                $optionName,
+                $filteredRawOptions[$optionName],
+            ),
+            array_keys($filteredRawOptions),
+        );
+    }
+
+    /**
+     * @param string|bool|int|float|array<string|bool|int|float|null>|null $value
+     */
+    private function unparseOption(
+        InputOption $option,
+        string $name,
+        array|bool|float|int|string|null $value,
+    ): string {
+        return match (true) {
+            $option->isNegatable() => \sprintf('--%s%s', $value ? '' : 'no-', $name),
+            !$option->acceptValue() => \sprintf('--%s', $name),
+            $option->isArray() => implode('', array_map(fn ($item) => $this->unparseOptionWithValue($name, $item), $value)),
```

Comment:
shouldn't this use a space separator instead of an empty string ?
And shouldn't we return array options as multiple items in the unparsed list ? If you want to use this as input to symfony/process for instance, you would need each array item as a separate process argument